docs/code/EntryPoints.md

```diff
@@ -0,0 +1,219 @@
+﻿# Entry Points And Helper Classes 
+
+## Overview
+
+Entry-points are a way to interface with ML.NET components, by specifying an execution graph of connected inputs and outputs of those components.
+Both the manifest describing available components and their inputs/outputs, and an "experiment" graph description, are expressed in JSON. 
+The recommended way of interacting with ML.NET through other programming languages is by composing, and exchanging pipeline or experiment graphs.  
+
+Through the documentaiton, we also refer to them as 'entry points nodes', and not just entry points, and that is because they are used as nodes of the experiemnt graphs. 
+The graph 'variables', the various values of the experiment graph JSON properties serve to describe the relationship between the entry point nodes. 
+The 'variables' are therefore the edges of the DAG. 
+
+All of ML.NET entry points are described by their manifest. The manifest is another JSON object that documents and describes the structure of an entry points. 
+Manifests are referenced to understand what an entry point does, and how it should be constructed, in a graph.  
+
+This document briefly describes the structure of the entry points, the structure of an entry point manifest, and mentions the ML.NET classes that help construct an entry point
+graph.
+
+## EntryPoint manifest - the definition of an entry point
+
+An example of an entry point manifest object, specifically for the MissingValueIndicator transform, is:
+
+```javascript
+{
+    "Name": "Transforms.ColumnTypeConverter",
+    "Desc": "Converts a column to a different type, using standard conversions.",
+    "FriendlyName": "Convert Transform",
+    "ShortName": "Convert",
+    "Inputs": [
+        {   "Name": "Column",
+            "Type": {
+                "Kind": "Array",
+                "ItemType": {
+                    "Kind": "Struct",
+                    "Fields": [
+                        {
+                            "Name": "ResultType",
+                            "Type": {
+                                "Kind": "Enum",
+                                "Values": [ "I1","I2","U2","I4","U4","I8","U8","R4","Num","R8","TX","Text","TXT","BL","Bool","TimeSpan","TS","DT","DateTime","DZ","DateTimeZone","UG","U16"]
```

Comment:
>"Values": [ "I1","I2","U2","I4","U4","I8","U8","R4","Num","R8","TX","Text","TXT","BL","Bool","TimeSpan","TS","DT","DateTime","DZ","DateTimeZone","UG","U16"] [](start = 32, length = 156)

Having judicious linebreaks is fine, just that one per element was a bit much.

Reply:
I condensed all the  '[ ] ' to be on the same line as the element. Most of the arrays contain one element. 

Keeping this in-line as well for consistency and to keep the graph shorter. I'll fix the spacing before/after the  '['']'to be consistent. 

---
In reply to: [194775271](https://github.com/dotnet/machinelearning/pull/295#discussion_r194775271) [](ancestors = 194775271)

Reply:
A comprehensible document understandable by its reader is the goal. Syntactic "consistency" **isn't a goal**. One way this can be incomprehensible is for it to be so long that the reader gets lost in the weeds, as was the case previously (and, frankly, is still the case). The other way it can be incomprehensible is to put everything on one line so structure can't be appreciated.

Think of it in these terms. If you were personally writing out this yourself, I doubt you would structure code in this way.

---
In reply to: [194776883](https://github.com/dotnet/machinelearning/pull/295#discussion_r194776883) [](ancestors = 194776883,194775271)